### ISSUE TITLE
False positive: let with flow annotation not counted as declaration

### ISSUE THREAD

**Issue by NilSet** [SonarSource/SonarJS#828]
**I want to report a bug.**
`let` variable declarations annotated with a flow type are seemingly not tracked as a declaration.

**SonarJS version:** 3.2<br>

**Rule key:** S2703 aka VariableDeclarationWithoutVarCheck

**Reproducer** 
```js
let unflowed;
unflowed = true; // ok

let flowed: boolean;
flowed = true; //false positive
```

**Expected behavior**
Recognize the annotated `let` statement as the declaration of `flowed` instead of assuming the usage of `flowed` is an implicit global declaration.
